tap_kpa/streams.py

```diff
@@ -116,3 +123,86 @@ def prepare_request_payload(self, context, next_page_token):
         payload = super().prepare_request_payload(context, next_page_token)
         payload.update({"response_id": context.get("response_id")})
         return payload
+
+
+class RolesListStream(KpaStream):
+    """Define custom stream."""
+
+    name = "roles"
+    path = "/roles.list"
+    records_jsonpath = "$.roles[*]"
+    rest_method = "POST"
+
+    schema = th.PropertiesList(
+        th.Property("id", th.StringType),
+        th.Property("name", th.StringType),
+    ).to_dict()
+
+    def prepare_request_payload(self, context, next_page_token):
+        payload = super().prepare_request_payload(context, next_page_token)
```

Comment:
If you're not overriding the logic here then you don't need to include this method `prepare_request_payload` in each of your streams

Reply:
Fixed.